packages/runtime/src/runtimes/react/index.tsx

```diff
@@ -275,16 +272,10 @@ const ElementReference = memo(
     { elementReference }: ElementRefereceProps,
     ref: Ref<unknown>,
   ): JSX.Element {
-    const { error, data } = useQuery(ELEMENT_REFERENCE_GLOBAL_ELEMENT, {
-      variables: { id: elementReference.value },
-    })
-    const globalElementData = data?.globalElement?.data as ReactPage.ElementData | undefined
+    const globalElement = useGlobalElement(elementReference.value)
+    const globalElementData = globalElement?.data as ReactPage.ElementData | undefined
     const elementReferenceDocument = useDocument(elementReference.key)
 
-    if (error != null) {
-      return <FallbackComponent ref={ref as Ref<HTMLDivElement>} text="Something went wrong!" />
-    }
```

Comment:
We could update the hook to keep this error handling if we want to avoid changing behavior too much